cmd/notification-service/main.go

```diff
@@ -315,6 +315,7 @@ func main() {
 		Repository:       notificationRepo,
 		Webhook:          notificationWebhook,
 		FeatureConnector: entitlementConnRegistry.Feature,
+		Logger:           logger.WithGroup("notification"),
```

Comment:
Note that this causes all attribute keys passed to this logger to be prefixed with `notification.`. I don't have all the context, but I'm not sure that's desirable.

We used `logger.With(slog.String("subsystem", "notification"))` in other places.

Reply:
Thank you for pointing out, I have changed the code accordingly.

Reply:
I used the `WithGroup` deliberately here. Will revert this change.